tests/components/homematicip_cloud/test_init.py

```diff
@@ -1,154 +1,115 @@
 """Test HomematicIP Cloud setup process."""
 
-from unittest.mock import patch
+from asynctest import CoroutineMock, patch
 
 from homeassistant.components import homematicip_cloud as hmipc
 from homeassistant.setup import async_setup_component
 
-from tests.common import Mock, MockConfigEntry, mock_coro
+from tests.common import Mock, MockConfigEntry
 
 
 async def test_config_with_accesspoint_passed_to_config_entry(hass):
     """Test that config for a accesspoint are loaded via config entry."""
-    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
-        hmipc, "configured_haps", return_value=[]
-    ):
-        assert (
-            await async_setup_component(
-                hass,
-                hmipc.DOMAIN,
-                {
-                    hmipc.DOMAIN: {
-                        hmipc.CONF_ACCESSPOINT: "ABC123",
-                        hmipc.CONF_AUTHTOKEN: "123",
-                        hmipc.CONF_NAME: "name",
-                    }
-                },
-            )
-            is True
-        )
 
-    # Flow started for the access point
-    assert len(mock_config_entries.flow.mock_calls) >= 2
+    entry_config = {
+        hmipc.CONF_ACCESSPOINT: "ABC123",
+        hmipc.CONF_AUTHTOKEN: "123",
+        hmipc.CONF_NAME: "name",
+    }
+    # no config_entry exists
+    assert len(hass.config_entries.async_entries(hmipc.DOMAIN)) == 0
+    # no acccesspoint exists
+    assert not hass.data.get(hmipc.DOMAIN)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)